### ISSUE TITLE
React Navigation `material-bottom-tabs` breaks nested Stack Navigators

### ISSUE THREAD

**Issue by traviswimer** [callstack/react-native-paper#2999]
### Current behaviour

On Android, the tab content is not visible the second time the tab is viewed. This is when using `@react-navigation/material-bottom-tabs`, with a Stack Navigator nested within one of the tabs.

To reproduce:

1. Press the tab that contains the Stack Navigator. (The content will be visible at this point)
2. Press a different tab.
3. Press the first tab again, and nothing will be displayed.

_The React Navigation docs said I should open an issue on this repo, and I believe the bug is related to `@react-navigation/material-bottom-tabs`. Let me know if is not the correct repo for this issue._

### Expected behaviour

The content should always be displayed for the selected tab.

### Code sample

```javascript
import React from "react";
import { Button, Text, View, StyleSheet } from "react-native";
import { NavigationContainer } from "@react-navigation/native";
import { createMaterialBottomTabNavigator } from "@react-navigation/material-bottom-tabs";
import { createNativeStackNavigator } from "@react-navigation/native-stack";

const Tab = createMaterialBottomTabNavigator();
const Stack = createNativeStackNavigator();

// Top level "Bottom Tab" screens
function FirstTabScreen() {
  // THIS IS THE TAB THAT DOESN'T DISPLAY CORRECTLY
  return (
    <Stack.Navigator initialRouteName="StackOne">
      <Stack.Screen name="StackOne" component={StackScreenOne} />
      <Stack.Screen name="StackTwo" component={StackScreenTwo} />
    </Stack.Navigator>
  );
}
function SecondTabScreen() {
  return (
    <View style={styles.screen}>
      <Text>Second Tab!</Text>
    </View>
  );
}

// Inner "Stack" screens
function StackScreenOne({ navigation }) {
  return (
    <View style={styles.screen}>
      <Text>Stack One!</Text>
      <Button
        title="Go to Stack Screen Two"
        onPress={() => {
          navigation.navigate("StackTwo");
        }}
      />
    </View>
  );
}
function StackScreenTwo({ navigation }) {
  return (
    <View style={styles.screen}>
      <Text>Stack Two!</Text>
      <Button
        title="Go to Stack Screen One"
        onPress={() => {
          navigation.navigate("StackOne");
        }}
      />
    </View>
  );
}

export default function App() {
  return (
    <NavigationContainer>
      <Tab.Navigator initialRouteName="FirstTab">
        <Tab.Screen
          name="FirstTab"
          component={FirstTabScreen}
          options={{
            tabBarLabel: "FirstTab",
            tabBarIcon: "home-outline",
          }}
        />
        <Tab.Screen
          name="SecondTab"
          component={SecondTabScreen}
          options={{
            tabBarLabel: "SecondTab",
            tabBarIcon: "bell-outline",
          }}
        />
      </Tab.Navigator>
    </NavigationContainer>
  );
}

const styles = StyleSheet.create({
  screen: { flex: 1, justifyContent: "center", alignItems: "center" },
});
```

### Screenshots (if applicable)

N/A

### What have you tried

- I tested on iOS and the bug is not present.
- I downgraded to version `5.3.19` and the bug was no longer present.
- I found an [old stackoverflow question](https://stackoverflow.com/a/49316722/1316229) that sounds similar, so maybe the same bug has returned?
    - I tried setting a height/width on the "stack" screens using `screenOptions` on the `Stack.Navigator`, but it had no effect.

### Your Environment

| software              | version
| --------------------- | -------
| android        | 12 (real device, I have not tested on an emulator)
| react-native          | 0.64.3
| react-native-paper    | 4.10.1
| node                  | 14.18.2
| npm or yarn           | yarn
| expo sdk              | 43.0.2

### Snacky Snack

https://snack.expo.dev/AWE08jYKr

**Comment by traviswimer** [comment]
It looks like I did in fact post this to the correct repo, but it is a duplicate of an issue on another repo: https://github.com/software-mansion/react-native-screens/issues/1197

But the good news is that the bug was already fixed before I even posted my issue, it's just not released yet. Now that's efficiency! 😎

**Comment by lukewalczak** [comment]
Hello @traviswimer could you please point me to the place where the issue was fixed?

**Comment by traviswimer** [comment]
@lukewalczak Unfortunately, I think I may have spoken too soon. I'm not sure how I got confused, but it looks like the commit I saw was actually just on a fork: https://github.com/Gruak/react-native-paper/commit/15d204a8fcd99d9b6459d54e7dab7904aba6b9f4

I'm not sure if they made a pull request or if it's actually reasonable for the library as a whole, but at least it can temporarily be fixed with a fork until a real fix is merged.

**Comment by lukewalczak** [comment]
@traviswimer thanks, I had the feeling like I'm missing something. I will investigate if we can include that change from the fork into library.

**Comment by amirfrd** [comment]
this issue exist on android

**Comment by traviswimer** [comment]
@amirfrd if you haven't figured it out yet, this bug has apparently been fixed if you upgrade to `react-native` >0.65.0 (I haven't personally tested this yet though)

There is also a simple fix if you don't want to upgrade:
 https://github.com/software-mansion/react-native-screens/issues/1197#issuecomment-993682256
 
 Basically just wrap your stack navigator with `<View style={{flex: 1}} collapsable={false}></View>`
